### ISSUE TITLE
Influence kernel parameters at install time

### ISSUE THREAD

**Issue by davidcassany** [rancher/elemental-cli#297]
Elemental client `elemental install` command should have some way to easily influence the kernel parameters within the bootloader.

Ideas:
* make the grub2 configuration file a template and render it at install phase
* pass kernel parameters as a grub environment variable that gets filled at install time
* patch grub.cfg at install time
* mix of template and grub2 approach

This will allow us to set all labels at install time and remove any partition or filesystem label literal from `elemental-toolkit` repository.

**Comment by mudler** [comment]
I wonder if we can't reuse already the grub extra boot params envs instead of generating the grub.cfg.

I've a mixed feeling about generating grub configs, at that point I feel we could drift away and show discrepancies between nodes.

**Comment by mudler** [comment]
Having a post-install hook seems also something that is already doable by using the `after-install` stage. so this is completely doable already by pushing a cloudconfig which alters the boot options, among the lines of https://rancher.github.io/elemental-toolkit/docs/customizing/configure_grub/#grub-environment-variables

**Comment by mudler** [comment]
This sounds really invasive and hard to maintain in the long run - do we need custom partition labels for state, persistency, and all? It feels reimplementing tools to generate grub configs.

So.. gotta ask: what's the motivation behind this?

If it comes to add boot parameters after install that's easily doable already by writing the grub env files, this can be easily instrumented from the agent too.

**Comment by kkaempf** [comment]
It's mostly about any quirks your hardware/driver combination has and you need to work around by passing kernel parameters (for ISO boot and for disk boot)

**Comment by mudler** [comment]
Right, but is this card about generating grub.cfg during install? That's a big design change as it starts to add dynamic config generation on runtime.

For such options on the livecd is not needed, neither after install. I'm just concerned about dynamic configuration generation as it propagates down to other paths, e.g. when resetting a system
